moncli/types.py

```diff
@@ -211,4 +214,78 @@ def validate_timezone(self, value):
         try:
             pytz.timezone(value)
         except (UnknownTimeZoneError):
-            raise ValidationError('Unknown time zone "{}".'.format(value))
\ No newline at end of file
+            raise ValidationError('Unknown time zone "{}".'.format(value))
+
+class StatusType(MondayType):
+    native_type = str
+    allow_casts = (int, str)
+    null_value = {}
+    def __init__(self, id: str = None, title: str = None,data_mapping:dict =None ,  *args, **kwargs):
+        if data_mapping:
+            values= [value for value in data_mapping.values()]        
+            self.native_type = values[0].__class__
+            self.choices = values
+        self._data_mapping =  data_mapping 
+        super().__init__(id=id, title=title, *args, **kwargs)
+    
+    def _process(self, value):
+        labels = self.metadata['labels']
+        if self.native_type == str:
+            try:
+                if isinstance(int(value),int):
+                    if value in labels.values():
```

Comment:
Lines 236-239 are not the correct way to check this.  If the value is a str that can be converted to an int, use the value as a key to pull out the label value from your _labels_ dict.  If that throws a KeyError, then you raise a ConversionError indicating that the label could not be found from the passed in index.